retrieval/target.go

```diff
@@ -196,11 +198,14 @@ func (t *Target) Update(cfg *config.ScrapeConfig, baseLabels, metaLabels clientm
 	if cfg.BasicAuth != nil {
 		t.url.User = url.UserPassword(cfg.BasicAuth.Username, cfg.BasicAuth.Password)
 	}
+	t.url.RawQuery = cfg.Params.Encode()
```

Comment:
It gets tricky though as a configurable separator won't suffice. If the query param is a metric selector it may contain arbitrary characters.
Also query params may contain non-alphanum chars and thus we don't get a valid labelname.